include/ppx/scene/scene_material.h

```diff
@@ -0,0 +1,325 @@
+    float         GetEmissiveStrength() const { return mEmissiveStrength; }
+
+    const scene::TextureView& GetBaseColorTextureView() const { return mBaseColorTextureView; }
+    const scene::TextureView& GetMetallicRoughnessTextureView() const { return mMetallicRoughnessTextureView; }
+    const scene::TextureView& GetNormalTextureView() const { return mNormalTextureView; }
+    const scene::TextureView& GetOcclusionTextureView() const { return mOcclusionTextureView; }
+    const scene::TextureView& GetEmissiveTextureView() const { return mEmissiveTextureView; }
+
+    scene::TextureView* GetBaseColorTextureViewPtr() { return &mBaseColorTextureView; }
+    scene::TextureView* GetMetallicRoughnessTextureViewPtr() { return &mMetallicRoughnessTextureView; }
+    scene::TextureView* GetNormalTextureViewPtr() { return &mNormalTextureView; }
+    scene::TextureView* GetOcclusionTextureViewPtr() { return &mOcclusionTextureView; }
+    scene::TextureView* GetEmissiveTextureViewPtr() { return &mEmissiveTextureView; }
+
+    bool HasBaseColorTexture() const { return mBaseColorTextureView.HasTexture(); }
+    bool HasMetallicRoughnessTexture() const { return mMetallicRoughnessTextureView.HasTexture(); }
+    bool HasNormalTexture() const { return mNormalTextureView.HasTexture(); }
+    bool HasOcclusionTexture() const { return mOcclusionTextureView.HasTexture(); }
+    bool HasEmissiveTexture() const { return mEmissiveTextureView.HasTexture(); }
+
+    void SetBaseColorFactor(const float4& value);
+    void SetMetallicFactor(float value);
+    void SetRoughnessFactor(float value);
+    void SetOcclusionStrength(float value);
+    void SetEmissiveFactor(const float3& value);
+    void SetEmissiveStrength(float value);
+
+private:
+    float4             mBaseColorFactor              = float4(1, 1, 1, 1);
+    float              mMetallicFactor               = 1;
+    float              mRoughnessFactor              = 1;
+    float              mOcclusionStrength            = 1;
+    float3             mEmissiveFactor               = float3(0, 0, 0);
+    float              mEmissiveStrength             = 0;
+    scene::TextureView mBaseColorTextureView         = {};
+    scene::TextureView mMetallicRoughnessTextureView = {};
+    scene::TextureView mNormalTextureView            = {};
+    scene::TextureView mOcclusionTextureView         = {};
+    scene::TextureView mEmissiveTextureView          = {};
+};
+
+// -------------------------------------------------------------------------------------------------
+
+// Material Factory
+//
+// Customizable factory that provides implementations of materials. An application
+// can inherit this class to provide implementations of materials as it sees fit.
+//
+// Materials must be uniquely identifiable by their Material Ident string.
+//
+// Materials that do not take any parameters, such as the default ErrorMaterial
+// class, can the same shared copy across all of its instances.
```

Comment:
can -> get no?

Reply:
Removed since this comment doesn't apply anymore.